server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -263,11 +264,7 @@ public TemporalAccessor parse(String input) {
             throw new IllegalArgumentException("cannot parse empty date");
         }
 
-        try {
-            return doParse(input);
-        } catch (DateTimeParseException e) {
-            throw new IllegalArgumentException("failed to parse date field [" + input + "] with format [" + format + "]", e);
-        }
+        return doParse(input);
```

Comment:
To respect the document in server/src/main/java/org/opensearch/common/time/DateFormatter.java, there is no need to catch DateTimeParseException and re-throw IllegalArgumentException.

Reply:
Why we are changing that?

Reply:
It's not supposed to throw IllegalArgumentException according to the  document in DateFormatter, you also had comments on the unit test and pointed out it should throw DateTimeParseException.

Reply:
Ah ... you are totally and absolutely correct (since it implements `JavaDateFormatter` impelements `DateFormatter`).

Reply:
I am really afraid that changing the exception type here may have unintended consequences, moreover you have updated the javadocs for `TemporalAccessor DateFormatter::parse(String input)`: it looks to me it has only ever throws `IllegalArgumentException` and not  `DateTimeParseException` (that was copy pasted from `DateTimeFormatter::parse(String input)`)

Reply:
You are right, it is too risky to change the behavior of it, I have search the code where the  DateFormatter is used, most of them won't be affect at all, but it's hard to tell it won't cause any other problem since it is being used in many places.

I have reverted the change and only updated the document on DateFormatter::parse.